File: TC_pruebas_login/test_playwright/tc_01.py
from playwright.sync_api import sync_playwright
from var import *

counter = 1

def tomar_screenshot(page, screen_name: str):
    global counter
    page.screenshot(path=f"TC_pruebas_login/test_playwright/test_images/pruebas_01_{screen_name}_{counter}.png")
    counter += 1

with sync_playwright() as p:
    browser = p.chromium.launch(headless=False) 
    page = browser.new_page()

    page.goto(link)
    page.wait_for_timeout(3000) 
    tomar_screenshot(page, "inicio_exitoso")

    page.locator(uname_input).fill("tomsmith")
    page.locator(pw_input).fill("SuperSecretPassword!")
    tomar_screenshot(page, "credenciales")

    page.locator(btn_continue).click()
    page.wait_for_timeout(2000)

    try:
        success_message = page.locator('//*[@id="content"]/div/h4')
        success_text = success_message.text_content()
        tomar_screenshot(page, "resultado")

        if "Welcome to the Secure Area. When you are done click logout below." in success_text:
            print(f"Test exitoso: {success_text}")
        else:
            print("Test fallido: Mensaje de éxito no encontrado")

    except Exception as e:
        print(f"Test fallido: Ocurrió un error - {e}")

    browser.close()

File: TC_pruebas_login/test_puppetter/tc_01.js
const puppeteer = require('puppeteer');

// Variables
const link = "https://the-internet.herokuapp.com/login";
const uname_input = '#username';
const pw_input = '#password';
const btn_continue = '#login button i';

let counter = 1;

async function tomar_screenshot(page, screen_name) {
    await page.screenshot({
        path: `test_images/pruebas_01_${screen_name}_${counter}.png`
    });
    counter++;
}

async function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

(async () => {
    const browser = await puppeteer.launch({ headless: false });
    const page = await browser.newPage();

    await page.goto(link, { waitUntil: 'networkidle2' });
    await sleep(3000);
    await tomar_screenshot(page, "inicio_exitoso");

    // Username
    await page.type(uname_input, "tomsmith");

    // Password
    await page.type(pw_input, "SuperSecretPassword!");
    await tomar_screenshot(page, "credenciales");

    // Botón login
    await page.click(btn_continue);
    await sleep(2000);

    try {
        const success_message = await page.$('#content h4');
        if (success_message) {
            const success_text = await page.evaluate(el => el.textContent, success_message);
            await tomar_screenshot(page, "resultado");

            if (success_text.includes("Welcome to the Secure Area. When you are done click logout below.")) {
                console.log(`✅ Test exitoso: ${success_text}`);
            } else {
                console.log("❌ Test fallido: Mensaje de éxito no encontrado");
            }
        } else {
            console.log("❌ Test fallido: No se encontró el mensaje de éxito");
        }
    } catch (e) {
        console.log(`❌ Test fallido: Ocurrió un error - ${e}`);
    }

    await browser.close();
})();
